src/test/scala/io/moia/protos/teleproto/ReaderTest.scala

```diff
@@ -151,5 +151,22 @@ class ReaderTest extends UnitTest {
       discountedReader.read(proto) shouldBe PbSuccess(false)
       discountedReader.read(proto.copy(discounts = Map("2" -> "0.5", "3" -> "0.8"))) shouldBe PbSuccess(true)
     }
+
+    "succeed with a derived sequenceReader" in {
+      implicit val r     = reader
+      val sequenceReader = implicitly[Reader[Seq[Protobuf], Seq[Model]]]
+      val result         = sequenceReader.read(Seq(proto, proto.copy(id = Some("bar"))))
```

Comment:
Thanks for your contribution!
Honestly, I wouldn't know why to do this `implicitly` and stuff if I could just do `Seq(...).map(reader.read)`.
Do you have a use case?

Reply:
Mmh but that would result in a `Seq[PbResult[StopFlag]]`, instead of a `PbResult[Seq[StopFlag]]`, right? Maybe there's another simple way to do this with some combinators, but I for one didn't manage it...

We have a use case in [this PR](https://github.com/moia-dev/stopresolver/pull/670#discussion_r1219194279) (code now moved to `src/main/scala/io/moia/stopresolver/stop/StopNetworkReader.scala`).

Reply:
The combinator you are looking for is called __traverse__. 😉